### PR TITLE
Update one URL rewrite rule as a test

### PR DESCRIPTION
**Pull request by SpiritLevel**
If OK, can someone merge this so I can check that all is well.

### CODE REVIEW

**Comment by camfindlay** [comment]
In this case we know it is a specific URL that is causing the 404. So I would have expected something more like

RewriteRule ^/changelog/?$ /en/changelogs [R=301,L]

The ?$ allows the incoming url to include or not the / at the end. By not including the version number in the destination, in the future when people hit the old link they'll get the latest stable version of the change logs.

Also watch adding lots of extra blank lines ;)

**Comment by camfindlay** [comment]
no need for the /3.1/ bit @SpiritLevel

**Comment by camfindlay** [comment]
Will test it on my local dev then you should be good, get the first one down so you know you're on the right track then go for gold.

File: .htaccess [hunk 1]
```diff
@@ -119,6 +122,7 @@
 	RewriteRule ^(.*)/topics/authentication$ /$1/developer_guides/security/authentication [R=301,L]
 	RewriteRule ^(.*)/topics/caching$ /$1/developer_guides/performance/caching [R=301,L]
 	RewriteRule ^(.*)/topics/commandline$ /$1/developer_guides/cli [R=301,L]
+	RewriteRule ^/topics/datamodel/?$ /en/developer_guides/model/data_model_and_orm  [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: Any 404s that include the /en/ part will need to follow the regex format with ^(.*) etc and $1 in the destination.

Comment (SpiritLevel) [inline .htaccess]: Or, use ^([a-z]{2}) as on line 57/59? Do you have a preference ?

Question: how do we know whether or not we are adding rewrites on top of existing rewrites ? Might it be better to comment all these rewrite rules out, and start from scratch with a new top 50 of the 404s ?

Comment (camfindlay) [inline .htaccess]: Maybe follow the established convention for now, we could review and do a bulk improvement later?

We could do a separate review of rewrites looking for dups later, what is most important is actually having less 404s :)

File: .htaccess [hunk 2]
```diff
@@ -54,10 +56,11 @@
 	RewriteRule ^(.*)/howto/pagination$ /$1/developer_guides/templates/how_tos/pagination [R=301,L]
 	RewriteRule ^(.*)/howto/simple-contact-form$ /$1/developer_guides/forms/how_tos/simple_contact_form [R=301,L]
 	
-	RewriteRule ^([a-z]{2})/installation/?$ /$1/getting_started/installation [R=301,L]
+	RewriteRule ^(.*)/installation/?$ /$1/getting_started/installation [R=301,L]
 	RewriteRule ^(.*)/installation/common-problems$ /$1/getting_started/installation/common_problems [R=301,L]
 	RewriteRule ^(.*)/installation/composer$ /$1/getting_started/composer [R=301,L]
 	RewriteRule ^(.*)/installation/from-source$ /$1/getting_started/installation [R=301,L]
+	RewriteRule ^(.*)/installation-into-subversion/?* /$1/getting_started/composer/  [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: See the links at #50 this one doesn't have a /en/ slug in it so we don't need (.*)

File: .htaccess [hunk 3]
```diff
@@ -54,10 +56,11 @@
 	RewriteRule ^(.*)/howto/pagination$ /$1/developer_guides/templates/how_tos/pagination [R=301,L]
 	RewriteRule ^(.*)/howto/simple-contact-form$ /$1/developer_guides/forms/how_tos/simple_contact_form [R=301,L]
 	
-	RewriteRule ^([a-z]{2})/installation/?$ /$1/getting_started/installation [R=301,L]
+	RewriteRule ^(.*)/installation/?$ /$1/getting_started/installation [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: question: which top 50 404 is this one fixing?

File: .htaccess [hunk 4]
```diff
@@ -37,6 +37,8 @@
 	RewriteRule ^(.*)/topics/testing/create-silverstripe-test$ /$1/topics/testing/creating-a-silverstripe-test [R=301,L]
 	RewriteCond %{REQUEST_URI} !(3.0|2.4)
 	RewriteRule ^(.*)/topics/testing/create-functional-test$ /$1/topics/testing/creating-a-functional-test [R=301,L]
+	
+        RewriteRule ^(.*)/changelog/?$ /$1/changelogs [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: no need for (.*) or $1 this is a direct rewrite, no regex required here.

File: .htaccess [hunk 5]
```diff
@@ -119,6 +122,7 @@
 	RewriteRule ^(.*)/topics/authentication$ /$1/developer_guides/security/authentication [R=301,L]
 	RewriteRule ^(.*)/topics/caching$ /$1/developer_guides/performance/caching [R=301,L]
 	RewriteRule ^(.*)/topics/commandline$ /$1/developer_guides/cli [R=301,L]
+	RewriteRule ^(.*)/topics/datamodel/?$ /$1/developer_guides/model/data_model_and_orm  [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: This one is all good :+1:

File: .htaccess (hunk 3 above)
Comment (SpiritLevel) [inline .htaccess]: This one was already in .htaccess. I only changed the [a-z]{2} to (.*) for consistency :)

Comment (camfindlay) [inline .htaccess]: ignore me in that case :)

File: .htaccess (hunk 4 above)
Comment (SpiritLevel) [inline .htaccess]: OK :)

**Comment by camfindlay** [comment]
Tidy up that last few comments and I'll look to merge and get on our testing server to ensure the rewrites work.

File: .htaccess (hunk 3 above)
Comment (SpiritLevel) [inline .htaccess]: I would never do that ;)

File: .htaccess [hunk 6]
```diff
@@ -54,10 +56,11 @@
 	RewriteRule ^(.*)/howto/pagination$ /$1/developer_guides/templates/how_tos/pagination [R=301,L]
 	RewriteRule ^(.*)/howto/simple-contact-form$ /$1/developer_guides/forms/how_tos/simple_contact_form [R=301,L]
 	
-	RewriteRule ^([a-z]{2})/installation/?$ /$1/getting_started/installation [R=301,L]
+	RewriteRule ^(.*)/installation/?$ /$1/getting_started/installation [R=301,L]
 	RewriteRule ^(.*)/installation/common-problems$ /$1/getting_started/installation/common_problems [R=301,L]
 	RewriteRule ^(.*)/installation/composer$ /$1/getting_started/composer [R=301,L]
 	RewriteRule ^(.*)/installation/from-source$ /$1/getting_started/installation [R=301,L]
+	RewriteRule ^/installation-into-subversion/?$ /getting_started/composer/  [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: drop the / after ^ for any that dont have a /en/ slug

File: .htaccess [hunk 7]
```diff
@@ -37,6 +37,8 @@
 	RewriteRule ^(.*)/topics/testing/create-silverstripe-test$ /$1/topics/testing/creating-a-silverstripe-test [R=301,L]
 	RewriteCond %{REQUEST_URI} !(3.0|2.4)
 	RewriteRule ^(.*)/topics/testing/create-functional-test$ /$1/topics/testing/creating-a-functional-test [R=301,L]
+	
+        RewriteRule ^/changelog/?$ /changelogs [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: and the / in this one....

**Comment by SpiritLevel** [comment]
Sorry...missed that last one....I guess I am not so good at multitasking. When I sit down to do them all at once, I'll be more careful ;)

File: .htaccess [hunk 8]
```diff
@@ -37,6 +37,8 @@
 	RewriteRule ^(.*)/topics/testing/create-silverstripe-test$ /$1/topics/testing/creating-a-silverstripe-test [R=301,L]
 	RewriteCond %{REQUEST_URI} !(3.0|2.4)
 	RewriteRule ^(.*)/topics/testing/create-functional-test$ /$1/topics/testing/creating-a-functional-test [R=301,L]
+	
+        RewriteRule ^changelog/?$ /changelogs [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: RewriteRule ^changelog/?$ /en/changelogs [R=301,L]

Comment (SpiritLevel) [inline .htaccess]: Sorry, Cam, that was a bit of a clumsy
      start on my part! Distracted at the moment.
      
      Hopefully these can be tested now. 
      
      Will do rewrites over weekend.
      
      Cheers,
      
      David.
      
      
      On 09/07/15 16:13, Cam Findlay wrote:
    
    
      In .htaccess:
      > @@ -37,6 +37,8 @@
>  	RewriteRule ^(.*)/topics/testing/create-silverstripe-test$ /$1/topics/testing/creating-a-silverstripe-test [R=301,L]
>  	RewriteCond %{REQUEST_URI} !(3.0|2.4)
>  	RewriteRule ^(.*)/topics/testing/create-functional-test$ /$1/topics/testing/creating-a-functional-test [R=301,L]
> +	
> +        RewriteRule ^changelog/?$ /changelogs [R=301,L] 

      RewriteRule ^changelog/?$ /en/changelogs [R=301,L]
      —
        Reply to this email directly or view
          it on GitHub.
      
        
          
          
        
        
      
    
    
    
    --

File: .htaccess [hunk 9]
```diff
@@ -54,10 +56,11 @@
 	RewriteRule ^(.*)/howto/pagination$ /$1/developer_guides/templates/how_tos/pagination [R=301,L]
 	RewriteRule ^(.*)/howto/simple-contact-form$ /$1/developer_guides/forms/how_tos/simple_contact_form [R=301,L]
 	
-	RewriteRule ^([a-z]{2})/installation/?$ /$1/getting_started/installation [R=301,L]
+	RewriteRule ^(.*)/installation/?$ /$1/getting_started/installation [R=301,L]
 	RewriteRule ^(.*)/installation/common-problems$ /$1/getting_started/installation/common_problems [R=301,L]
 	RewriteRule ^(.*)/installation/composer$ /$1/getting_started/composer [R=301,L]
 	RewriteRule ^(.*)/installation/from-source$ /$1/getting_started/installation [R=301,L]
+	RewriteRule ^installation-into-subversion/?$ /getting_started/composer/ [R=301,L]
```
Comment (camfindlay) [inline .htaccess]: one more /en/ missing here... 

RewriteRule ^installation-into-subversion/?$ /en/getting_started/composer/ [R=301,L]

**Comment by camfindlay** [comment]
I'll leave you with another challenge to look into... squash your commits in git into a single commit for the pull request ;) 

Experiential learning via GitHub :+1:

**Comment by SpiritLevel** [comment]
Lots to be squashed too!

**Comment by dhensby** [comment]
:(

**Comment by SpiritLevel** [comment]
Worry not. These changes went in. It took a few tries to get the squashing working.

**Pull request closed without merging**